src/dimension.c

```diff
@@ -970,10 +1046,18 @@ ts_hyperspace_calculate_point(const Hyperspace *hs, TupleTableSlot *slot)
 		bool isnull;
 		Oid dimtype;
 
+		AttrNumber partition_col_idx = d->column_attno;
+		if (colnames_list)
+			partition_col_idx = get_partition_attrno_from_tuple(hs,
+																slot,
+																colnames_list,
+																(d->partitioning ? true : false));
```

Comment:
This is not the correct way to determine the dimension type. There are two macros IS_OPEN_DIMENSION and IS_CLOSED_DIMENSION to determine dimension type.

Reply:
Here iam not determining dimension type. 
For a given tuple received from subplan of ChunkDispatch, I need to know the position where time or space dimension column is located. get_partition_attrno_from_tuple() will return the slot number where the time or space dimension column is present in the tuple.